### ISSUE TITLE
Service becomes non-responsive when both query params n & filter are provided

### ISSUE THREAD

**Issue by Kaustubh-cribl** [NguyenIconAI/logspullerservice#2]
Problem description:

When both query params, n & filter, are provided, service becomes non-responsive for large files (more than 1 gig)

Acceptance criteria:

API should return n matching lines.

![Screenshot 2024-06-05 at 10 26 03 AM](https://github.com/NguyenIconAI/logspullerservice/assets/171298919/5d5a643f-51e7-40d0-90f8-748c65e65a82)
![Screenshot 2024-06-05 at 10 24 07 AM](https://github.com/NguyenIconAI/logspullerservice/assets/171298919/5d4f5ce9-6f25-404c-ac8c-dfbfb2ee48a6)

**Comment by NguyenIconAI** [comment]
It is actually a TODO item 
```
if line != "" {
        // TODO - Optimize the contains check while gathering the buffer
        if filter == "" || strings.Contains(line, filter) {
                lines = append([]string{line}, lines...)
        }
        line = ""
}
```
Tried to avoid premature optimization but well...

**Comment by NguyenIconAI** [comment]
issue is reproducible with:

```
func Benchmark_ReadLastNLines_NotFound(b *testing.B) {
	for _, v := range table {
		b.Run(fmt.Sprintf("input_size_%d", v.SizeInMB), func(b *testing.B) {
			logFilePath := "large_access.log"
			err := generateLargeLog(logFilePath, targetSizeMB)
			if err != nil {
				b.Fatalf("Failed to generate log file: %v", err)
			}

			b.ResetTimer()
			for n := 0; n < b.N; n++ {
				_, err := ReadLastNLines(logFilePath, 1000, "iddqd")
				if err != nil {
					b.Fatalf("Failed to read last N lines: %v", err)
				}
			}
		})
	}
}
```

```
$  go test ./... -bench=. -timeout=10s;
```

**Comment by NguyenIconAI** [comment]
This issue is fixed in commit ae4dafe2d6b2f08935a79ed4d4c3bde3215700d0

**Comment by NguyenIconAI** [comment]
Please verify the issue
